### ISSUE TITLE
Run error on Windows

### ISSUE THREAD

**Issue by ryanbuiltthat** [iron-meteor/iron-cli#138]
I'm trying to run my app using the `iron run` command or just `iron` but I'm hitting an error.

`..\npm\node_modules\iron-meteor\node_modules\fibers\future.js:278`
`Error: spawn Unknown system errno 193`

I'm just going to go ahead and say it, I have no effin' clue what this means. :) The app seems to do ok if I `cd` into the app directory and just run meteor. Am I missing something here?

**Comment by cmather** [comment]
Hmm does the nodes spawn command not work the same way on windows maybe? I may have to open up a VM and try it out.

**Comment by ryanbuiltthat** [comment]
Starting with meteor for me is also starting with node unfortunately. I didn't get any warnings about compatibility but you gave me at least a place to look into. Thanks! Not sure how much longer I'll keep my eyes open on this tonight but if I come across anything I'll let you know. 

For the record I'm using the latest meteor package 1.1.0.1 on Win8.1

**Comment by cmather** [comment]
Yeah keep me posted. Will keep a mental note to check this out on next cycle.

**Comment by trinric** [comment]
I am getting a similar error when I try to run `iron run` : 

Running windows 7 npm version 2.7.4, node v0.12.2

`C:\Users\Ryan\AppData\Roaming\npm\node_modules\iron-meteor\bin\iron:12
}).run();
   ^
Error: spawn UNKNOWN
    at exports._errnoException (util.js:746:11)
    at ChildProcess.spawn (child_process.js:1162:11)
    at exports.spawn (child_process.js:995:9)
    at module.exports (C:\Users\Ryan\AppData\Roaming\npm\node_modules\iron-meteo
r\node_modules\shell-source\index.js:36:14)
    at syncSource (C:\Users\Ryan\AppData\Roaming\npm\node_modules\iron-meteor\li
b\commands\run.js:9:3)
    at Command.<anonymous> (C:\Users\Ryan\AppData\Roaming\npm\node_modules\iron-
meteor\lib\commands\run.js:30:5)
    at Command.invoke (C:\Users\Ryan\AppData\Roaming\npm\node_modules\iron-meteo
r\lib\command.js:111:26)
    at Command.run (C:\Users\Ryan\AppData\Roaming\npm\node_modules\iron-meteor\l
ib\command.js:91:21)
    at Command.runSubCommand (C:\Users\Ryan\AppData\Roaming\npm\node_modules\iro
n-meteor\lib\command.js:154:11)
    at Command.<anonymous> (C:\Users\Ryan\AppData\Roaming\npm\node_modules\iron-
meteor\lib\iron.js:66:17)`

**Comment by Danjavia** [comment]
Same here 
    c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\node_modules\fibers\future.js:278
                                                throw(ex);
                                                      ^
Error: spawn Unknown system errno 193
    at Object.Future.wait (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\node_modules\fibers\future.js:398:15)
    at syncSource (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\commands\run.js:10:17)
    at Command.<anonymous> (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\commands\run.js:30:5)
    at Command.invoke (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:111:26)
    at Command.run (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:91:21)
    at Command.runSubCommand (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:154:11)
    at Command.<anonymous> (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\iron.js:66:17)
    at Command.invoke (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:111:26)
    at Command.run (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:91:21)
    at c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\bin\iron:11:8
    - - - - -
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)

**Comment by Danjavia** [comment]
Occur on windows 7 
Thanks!

**Comment by ryanbuiltthat** [comment]
I've been hitting a few issues with some node modules and Windows lately. re-installing those modules specifically seems to help in a few of the cases. It may be worth a try to cd into the directory and run `npm instlall` and the specific module having an issue.

**Comment by Danjavia** [comment]
Trying...

$ npm install
npm WARN package.json hlabs@0.0.1 No repository field.

> hlabs@0.0.1 postinstall c:\Users\danjavia\Desktop\Danjavia
> bower install --config.interactive=false

danjavia@DIEGOSHK /C/Users/danjavia/Desktop/Danjavia/shrmelink
$ iron run

c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\node_modules\fibers\future.js:278
                                                throw(ex);
                                                      ^
Error: spawn Unknown system errno 193
    at Object.Future.wait (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\node_modules\fibers\future.js:398:15)
    at syncSource (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\commands\run.js:10:17)
    at Command.<anonymous> (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\commands\run.js:30:5)
    at Command.invoke (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:111:26)
    at Command.run (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:91:21)
    at Command.runSubCommand (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:154:11)
    at Command.<anonymous> (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\iron.js:66:17)
    at Command.invoke (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:111:26)
    at Command.run (c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\lib\command.js:91:21)
    at c:\Users\danjavia\AppData\Roaming\npm\node_modules\iron-meteor\bin\iron:11:8
    - - - - -
    at errnoException (child_process.js:988:11)
    at Process.ChildProcess._handle.onexit (child_process.js:779:34)

**Comment by panphora** [comment]
I'm getting the same error trying to run the `iron` command on Win7.

---

PS F:\GitHub\Artisfy> iron

C:\Users\Seeker\AppData\Roaming\npm\node_modules\iron-meteor\node_modules\fibers
\future.js:278
                                                throw(ex);
                                                      ^
Error: spawn Unknown system errno 193
    at Object.Future.wait (C:\Users\Seeker\AppData\Roaming\npm\node_modules\iron
-meteor\node_modules\fibers\future.js:398:15)
    at syncSource (C:\Users\Seeker\AppData\Roaming\npm\node_modules\iron-meteor\
lib\commands\run.js:10:17)
    at Command.<anonymous> (C:\Users\Seeker\AppData\Roaming\npm\node_modules\iro
n-meteor\lib\commands\run.js:30:5)
    at Command.invoke (C:\Users\Seeker\AppData\Roaming\npm\node_modules\iron-met
eor\lib\command.js:111:26)
    at Command.run (C:\Users\Seeker\AppData\Roaming\npm\node_modules\iron-meteor
\lib\command.js:91:21)
    at Command.runSubCommand (C:\Users\Seeker\AppData\Roaming\npm\node_modules\i
ron-meteor\lib\command.js:154:11)
    at Command.<anonymous> (C:\Users\Seeker\AppData\Roaming\npm\node_modules\iro
n-meteor\lib\iron.js:66:17)
    at Command.invoke (C:\Users\Seeker\AppData\Roaming\npm\node_modules\iron-met
eor\lib\command.js:111:26)
    at Command.run (C:\Users\Seeker\AppData\Roaming\npm\node_modules\iron-meteor
\lib\command.js:91:21)
    at C:\Users\Seeker\AppData\Roaming\npm\node_modules\iron-meteor\bin\iron:11:
8
    - - - - -
    at errnoException (child_process.js:1001:11)
    at Process.ChildProcess._handle.onexit (child_process.js:792:34)

**Comment by mcrayne** [comment]
I get the same error on Windows 8.1. (error no. 193).  Running Meteor 1.1.0.2. All latest Windows patches.

**Comment by mitzip** [comment]
I think this is because shell-source isn't windows compatible...
https://github.com/jessetane/shell-source#how

In addition to the default source script provided, forward slashes are hardcoded as directory separators in index.js...
https://github.com/jessetane/shell-source/blob/master/index.js#L21

**Comment by cmather** [comment]
Oops. That might be it. The purpose of that is to source the env.sh file. You can see the code in `lib/commands/run.js` line 29. Perhaps we should make a PR to shell-source or just do this another way.

**Comment by mitzip** [comment]
I like the idea of a PR to shell-source, but then again I'm bias as I just built my own PR for iron-meteor around shell-source features... Fixing the hard coded forward slash in their index.js and using my PR to expose a way to customize how env.sh is sourced could fix the problems for windows folks. 
https://github.com/iron-meteor/iron-cli/pull/142

**Comment by pavan4** [comment]
Is this fixed ? I tried your fix @mitzip  but it didn't work for me.

**Comment by Auryg** [comment]
Same issue here.

**Comment by Crystal2030** [comment]
Is anybody fixed this issue? I met the same issue.

**Comment by ryanbuiltthat** [comment]
I gave up a while ago and went with Flow Router. The idea of scaffolding
was nice and all but I just got better at writing generic templates I could
use repeatedly.

**Comment by babin93** [comment]
I also get this error, anybody knows if any fix is planned ?

**Comment by chrisbutler** [comment]
@ryanbuiltthat @Crystal2030 @Auryg @pavan4 @babin93 can you guys try to update to version 1.2.2 and see if that makes a difference?

**Comment by wodCZ** [comment]
@chrisbutler installed right now and found this issue. In v1.2.2 it is same:
![image](http://i.imgur.com/d7B0LpJ.png)

**Comment by chrisbutler** [comment]
hey everyone, hopefully this will fix the ```iron run``` issue, although we still need to figure out how to support sourcing ```env.sh``` on Windows

**Comment by cmather** [comment]
Nice work chris. I wonder if it's better to just have the settings.json be the primary source of info, and it can read from environment variables for production environments. Those environment variables could be set in any number of ways. This would require putting the settings.json file through a template parser. It might be something like this:

```javascript
{
  database_password: <%= process.env['DATABASE_PASSWORD'] %>
}
```

**Comment by spyhole** [comment]
E:\DevProjects>nvmw use 0.12.7
Now using node v0.12.7 x64

E:\DevProjects>iron version
1.3.0

E:\DevProjects>iron create test
created test
created test\.iron
created test\.iron\config.json
created test\bin
created test\bin\.gitkeep
created test\build
created test\build\.gitkeep
created test\build\README
created test\config
created test\config\.gitkeep
created test\config\development
created test\config\development\env.sh
created test\config\development\settings.json
created test\config\production
created test\config\production\env.sh
created test\config\production\settings.json
Creating project  app... |
Meteor app created
created test\app\client
created test\app\client\.gitkeep
created test\app\client\collections
created test\app\client\collections\.gitkeep
created test\app\client\head.html
created test\app\client\lib
created test\app\client\lib\.gitkeep
created test\app\client\stylesheets
created test\app\client\stylesheets\.gitkeep
created test\app\client\templates
created test\app\client\templates\.gitkeep
created test\app\client\templates\layouts
created test\app\client\templates\layouts\master_layout
created test\app\client\templates\layouts\master_layout\master_layout.html
created test\app\client\templates\layouts\master_layout\master_layout.js
created test\app\client\templates\shared
created test\app\client\templates\shared\loading
created test\app\client\templates\shared\loading\loading.html
created test\app\client\templates\shared\not_found
created test\app\client\templates\shared\not_found\not_found.html
created test\app\lib
created test\app\lib\.gitkeep
created test\app\lib\collections
created test\app\lib\collections\.gitkeep
created test\app\lib\controllers
created test\app\lib\controllers\.gitkeep
created test\app\packages
created test\app\packages\.gitkeep
created test\app\private
created test\app\private\.gitkeep
created test\app\public
created test\app\public\.gitkeep
created test\app\server
created test\app\server\.gitkeep
created test\app\server\bootstrap.js
created test\app\server\collections
created test\app\server\collections\.gitkeep
created test\app\server\lib
created test\app\server\lib\.gitkeep
created test\app\server\publish.js
created app\client\templates\home
created app\client\templates\home\home.html
created app\client\templates\home\home.js
created app\client\templates\home\home.css
created app\lib\controllers\home_controller.js
created app\lib\routes.js
created app\lib\methods.js
created app\server\methods.js
Installing the package  iron:router... \
Changes to your project's package version selections:

iron:controller        added, version 1.0.8
iron:core              added, version 1.0.8
iron:dynamic-template  added, version 1.0.8
iron:layout            added, version 1.0.8
iron:location          added, version 1.0.9
iron:middleware-stack  added, version 1.0.9
iron:router            added, version 1.0.9
iron:url               added, version 1.0.9

iron:router: Routing specifically designed for Meteor

✔ iron:router
created app\client\stylesheets\main.css

E:\DevProjects>cd test

E:\DevProjects\test>iron run
> meteor run --settings E:\DevProjects\test\config\development\settings.json
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: spawn meteor ENOENT
    at exports._errnoException (util.js:746:11)
    at Process.ChildProcess._handle.onexit (child_process.js:1053:32)
    at child_process.js:1144:20
    at process._tickCallback (node.js:355:11)
    at Function.Module.runMain (module.js:503:11)
    at startup (node.js:129:16)
    at node.js:814:3

**Comment by spyhole** [comment]
fixed.
see https://github.com/iron-meteor/iron-cli/pull/190

**Comment by chrisbutler** [comment]
@spyhole do you have another process running on the same port?

**Comment by chrisbutler** [comment]
thanks @spyhole. this fix is released in 1.3.1

**Comment by spyhole** [comment]
@chrisbutler windows bugs(#198&#199 #201) fixed here https://github.com/iron-meteor/iron-cli/pull/205
